nn_meter/dataset/gnn_dataloader.py

```diff
@@ -1,2 +1,285 @@
 # Copyright (c) Microsoft Corporation.
-# Licensed under the MIT license.
\ No newline at end of file
+# Licensed under the MIT license.
+import torch
+import jsonlines
+from nn_meter import download_from_url
+import os
+import logging
+import random
+import dgl
+
+RAW_DATA_URL = "https://github.com/microsoft/nn-Meter/releases/download/v1.0-data/datasets.zip"
+__user_dataset_folder__ = os.path.expanduser('~/.nn_meter/dataset')
+
+hws = [
+    "cortexA76cpu_tflite21",
+    "adreno640gpu_tflite21",
+    "adreno630gpu_tflite21",
+    "myriadvpu_openvino2019r2",
+]
+
+
+class GNNDataset(torch.utils.data.Dataset):
+    def __init__(self, data_dir='./dataset', train=True, device='cpu', split_ratio=0.8):
+        """
+        Dataloader of the Latency Dataset
+
+        Parameters
+        ----------
+        data_dir : string
+            Path to save the downloaded dataset
+        train: bool
+            Get the train dataset or the test dataset
+        device: string
+            The Device type of the corresponding latency
+        shuffle: bool
+            If shuffle the dataset at the begining of an epoch
+        batch_size: int
+            Batch size.
+        split_ratio: float
+            The ratio to split the train dataset and the test dataset.
+        """
+        err_str = "Only support device type cpu/gpu640/gpu630/vpu"
+        assert device in ['cpu', 'gpu640', 'gpu630', 'vpu'], err_str
```

Comment:
You can use variable `hws` to replace the hard code ['cpu', 'gpu640', 'gpu630', 'vpu'] and directly set the init parameter `device` as a string in `hws`. (Is there any problem could be caused in GNN?)